connection.go

```diff
@@ -245,7 +245,9 @@ func (c *connection) read() ([]byte, error) {
 	if err == nil {
 		return c.dataBuf, nil
 	} else {
-		c.setState(connInactive)
+		if !isTemporaryNetError(err) {
```

Comment:
Without checking if the error is for a temporary network error (like read / write timeout), the connection was marked `connInactive` which would cause the connection manager to never choose it out of the pool.